Network/Socket/Types.hsc

```diff
@@ -79,7 +87,7 @@ import Network.Socket.Imports
 -----------------------------------------------------------------------------
 
 -- | Basic type for a socket.
-data Socket = Socket (IORef CInt) CInt {- for Show -}
+data Socket = Socket !(IORef CInt) !CInt {- for Show -}
```

Comment:
Ah... actually, it should probably just be an `IORef Validity`! That removes the redundancy and also the `CInt` box in the `IORef`.